src/machine.rs

```diff
@@ -482,6 +481,21 @@ impl<'mir, 'tcx> MiriMachine<'mir, 'tcx> {
         let rng = StdRng::seed_from_u64(config.seed.unwrap_or(0));
         let borrow_tracker = config.borrow_tracker.map(|bt| bt.instanciate_global_state(config));
         let data_race = config.data_race_detector.then(|| data_race::GlobalState::new(config));
+        let page_size = if let Some(page_size) = config.page_size {
+            page_size
+        } else {
+            let target = &layout_cx.tcx.sess.target;
+            match target.arch.as_ref() {
+                "wasm32" | "wasm64" => 64 * 1024,
+                "aarch64" =>
+                    if target.options.vendor.as_ref().eq("apple") {
+                        16 * 1024
+                    } else {
+                        4 * 1024
+                    },
```

Comment:
Are there references you can link here that document these non-4K page sizes?

Reply:
I'll add some in, thanks. I took the values based on #2644

Reply:
Yeah before we enshrine this in code I'd like to see sources for these numbers. That issue contains a source for wasm which you can just use; I assume it can't be too hard to find a source for apple.

Reply:
Added some sources; are they fine? Seems like Apple never *definitively* said post-A8 iOS/macOS uses 16k pages in their docs